[BUG]: test_get_inforom_version fails on boards where Board Part Number is not programmed (e.g. NVIDIA T4G)

### Is this a duplicate?

- [x] I confirmed there appear to be no [duplicate issues](https://github.com/NVIDIA/cuda-python/issues) for this bug and that I agree to the [Code of Conduct](CODE_OF_CONDUCT.md)

### Type of Bug

Runtime Error

### Component

cuda.core

### Describe the bug

`cuda_core/tests/system/test_system_device.py::test_get_inforom_version` asserts `len(inforom.board_part_number) > 0`, but on some valid boards NVML returns an empty string for this field — the OEM did not program a board part number into the InfoROM. This is the same condition that `nvidia-smi -q` surfaces as `Board Part Number: N/A`. The Python binding (`cuda.bindings.nvml.device_get_board_part_number`) faithfully forwards NVML's behavior, so the empty string is a legitimate value and the test's assumption that this field is always populated is too strict. 

Per a maintainer comment, CI hardware coverage does not currently include such a board, which is why the assertion was not caught earlier.

### How to Reproduce

1. Use a board where Board Part Number is not programmed in InfoROM (e.g. NVIDIA T4G with Driver 595.45.04 and CUDA 13.2).
2. Confirm the board reports `N/A` via ` nvidia-smi -q | grep -i "Board Part Number" Board Part Number : N/A`
3. Run test case `pytest cuda_core/tests/system/test_system_device.py::test_get_inforom_version`
4. Observed: 
```shell
assert len(board_part_number) > 0
E AssertionError: 0 > 0 E + where 0 = len('')
```

### Expected behavior

The test should accept an empty string for `inforom.board_part_number` (matching `nvidia-smi -q`'s `Board Part Number: N/A`).

### Operating System

_No response_

### nvidia-smi output

_No response_

--- a/cuda_core/tests/system/test_system_device.py
+++ b/cuda_core/tests/system/test_system_device.py
@@ -547,7 +547,9 @@ def test_get_inforom_version():
         with unsupported_before(device, "HAS_INFOROM"):
             board_part_number = inforom.board_part_number
         assert isinstance(board_part_number, str)
-        assert len(board_part_number) > 0
+
+        # Some boards (e.g. NVIDIA T4G) do not program a board part number
+        assert board_part_number == "" or board_part_number.strip() == board_part_number
 
         inforom.validate()
 

